src/main/java/com/minecolonies/core/colony/buildings/AbstractBuilding.java

```diff
@@ -164,6 +164,11 @@ public abstract class AbstractBuilding extends AbstractBuildingContainer
      */
     private BlockPos cachedStandingPosition;
 
+    /**
+     * List of building flags.
+     */
+    protected List<String> buildingFlags = new ArrayList<>();
```

Comment:
Wouldn’t a Set make more sense here than a List?
I don’t believe the order would matter, just the existence of flags